server/src/services/commits.ts

```diff
@@ -113,6 +115,33 @@ const addCommit = async (
   );
 };
 
+/**
+ * Pays for the commit with the specified commitId.
+ * An error will be thrown if the commit is not SUCCESSFUL.
+ * @param commitId Id of the commit to be deleted
+ * @param paymentData Data of the payment request
+ */
+const payForCommit = async (
```

Comment:
I like how you put `payForCommit` in services and `editCommit` in storage, so storage functions are just simply CRUD and services have functions more specific to the business functions.